File: server/model/src/main/kotlin/hu/bsstudio/bssweb/member/model/Member.kt
package hu.bsstudio.bssweb.member.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import hu.bsstudio.bssweb.member.common.MemberStatus
import java.time.LocalDate
import java.util.UUID

data class Member @JsonCreator constructor(
    @JsonProperty("id") val id: UUID,
    @JsonProperty("url") val url: String,
    @JsonProperty("name") val name: String,
    @JsonProperty("nickname") val nickname: String,
    @JsonProperty("description") val description: String,
    @JsonProperty("joinedAt") val joinedAt: LocalDate,
    @JsonProperty("role") val role: String,
    @JsonProperty("status") val status: MemberStatus,
    @JsonProperty("archived") val archived: Boolean
)

File: integration/src/types/member.ts
import type { UUID } from './common.js';

export enum Status {
  MEMBER = 'MEMBER',
  MEMBER_CANDIDATE = 'MEMBER_CANDIDATE',
  MEMBER_CANDIDATE_CANDIDATE = 'MEMBER_CANDIDATE_CANDIDATE',
}

export interface CreateMember {
  url: string;
  name: string;
}

export interface UpdateMember extends CreateMember {
  nickname: string;
  description: string;
  joinedAt: string; // ISO date string
  role: string;
  status: Status;
  archived: boolean;
}

export interface Member extends UpdateMember {
  id: UUID;
}